src/Akeneo/Tool/Bundle/MeasureBundle/Application/CreateMeasurementFamily/CreateMeasurementFamilyHandler.php

```diff
@@ -21,22 +23,30 @@ class CreateMeasurementFamilyHandler
     /** @var MeasurementFamilyRepositoryInterface */
     private $measurementFamilyRepository;
 
-    public function __construct(MeasurementFamilyRepositoryInterface $measurementFamilyRepository)
-    {
+    private ?EventDispatcherInterface $eventDispatcher;
+
+    /** TODO: @pullup Remove = null */
+    public function __construct(
+        MeasurementFamilyRepositoryInterface $measurementFamilyRepository,
+        EventDispatcherInterface $eventDispatcher = null
+    ) {
         $this->measurementFamilyRepository = $measurementFamilyRepository;
+        $this->eventDispatcher = $eventDispatcher;
     }
 
     public function handle(CreateMeasurementFamilyCommand $saveMeasurementFamilyCommand): void
     {
         $units = $this->units($saveMeasurementFamilyCommand);
+        $measurementFamilyCode = MeasurementFamilyCode::fromString($saveMeasurementFamilyCommand->code);
         $measurementFamily = MeasurementFamily::create(
-            MeasurementFamilyCode::fromString($saveMeasurementFamilyCommand->code),
+            $measurementFamilyCode,
             LabelCollection::fromArray($saveMeasurementFamilyCommand->labels),
             UnitCode::fromString($saveMeasurementFamilyCommand->standardUnitCode),
             $units
         );
 
         $this->measurementFamilyRepository->save($measurementFamily);
+        $this->eventDispatcher->dispatch(new MeasurementFamilyCreated($measurementFamilyCode));
```

Comment:
Do not call dispatch function if eventDispatcher is null (Same for DeleteMeasurementFamily and SaveMeasurementFamily)

```suggestion
        if ($this->eventDispatcher != null) {
            $this->eventDispatcher->dispatch(new MeasurementFamilyCreated($measurementFamilyCode));
        }
```

Reply:
yes you are right! done!